Microsoft.HealthVault.Fhir/Transformers/ImmunizationToFhir.cs

```diff
@@ -0,0 +1,92 @@
+        {
+            return ImmunizationToFhir.ToFhirInternal(hvImmunization, hvImmunization.ToFhirInternal<Hl7.Fhir.Model.Immunization>());
+        }
+    }
+    internal static class ImmunizationToFhir
+    {
+        internal static Hl7.Fhir.Model.Immunization ToFhirInternal(ItemTypes.Immunization hvImmunization, Hl7.Fhir.Model.Immunization fhirImmunization)
+        {
+            fhirImmunization.VaccineCode = hvImmunization.Name.ToFhir();
+            fhirImmunization.DateElement = hvImmunization.DateAdministrated.ToFhir();
+
+            if (hvImmunization.Administrator != null)
+            {
+                fhirImmunization.Practitioner.Add(new Hl7.Fhir.Model.Immunization.PractitionerComponent
+                {
+                    Actor = AddContainedResource(fhirImmunization, hvImmunization.Administrator.ToFhir())
+                });
+            }
+
+            if (hvImmunization.Manufacturer != null)
+            {
+                fhirImmunization.Manufacturer = AddContainedResource(fhirImmunization, new Hl7.Fhir.Model.Organization
+                {
+                    Name = hvImmunization.Manufacturer.Text
+                });
+            }
+
+            fhirImmunization.LotNumber = hvImmunization.Lot;
+            fhirImmunization.Route = hvImmunization.Route.ToFhir();
+            fhirImmunization.ExpirationDateElement = hvImmunization.ExpirationDate.ToFhir();
+
+            if (!string.IsNullOrEmpty(hvImmunization.Sequence)
+                || !string.IsNullOrEmpty(hvImmunization.AdverseEvent)
+                || !string.IsNullOrEmpty(hvImmunization.Consent))
+            {
+                var immunizationExtension = new Extension
+                {
+                    Url = HealthVaultExtensions.ImmunizationDetail
+                };
+
+                if (!string.IsNullOrEmpty(hvImmunization.Sequence))
+                {
+                    immunizationExtension.AddExtension(HealthVaultExtensions.ImmunizationDetailSequence, new FhirString(hvImmunization.Sequence));
+                }
+                if (!string.IsNullOrEmpty(hvImmunization.Consent))
+                {
+                    immunizationExtension.AddExtension(HealthVaultExtensions.ImmunizationDetailConcent, new FhirString(hvImmunization.Consent));
+                }
+                if (!string.IsNullOrEmpty(hvImmunization.AdverseEvent))
+                {
+                    immunizationExtension.AddExtension(HealthVaultExtensions.ImmunizationDetailAdverseEvent, new FhirString(hvImmunization.AdverseEvent));
+                }
+                fhirImmunization.Extension.Add(immunizationExtension);
+            }
+
+            fhirImmunization.Site = hvImmunization.AnatomicSurface.ToFhir();
+            return fhirImmunization;
+        }
+
+        private static ResourceReference AddContainedResource(DomainResource domainResource, Resource resource, string id = null)
```

Comment:
This feels more like a global helper than a private method. I believe we're doing very similar things already in procedure and the approach could be re-used.